File: csharp/src/FizzBuzz.cs
namespace csharp;

public class FizzBuzz : IFizzBuzz
{
    public string Say(int number)
    {
        if (number % 15 == 0)
        {
            return "FizzBuzz";
        }

        if (number % 3 == 0)
        {
            return "Fizz";
        }

        if (number % 5 == 0)
        {
            return "Buzz";
        }

        return number.ToString();
    }
}

File: typescript/src/FizzBuzz.ts
import {IFizzBuzz} from "./IFizzBuzz";

export class FizzBuzz implements IFizzBuzz {
    public say(inputNumber: number): string {
        if (inputNumber % 15 === 0) {
            return "FizzBuzz";
        }
        if (inputNumber % 3 === 0) {
            return "Fizz";
        }
        if (inputNumber % 5 === 0) {
            return "Buzz";
        }

        return String(inputNumber);
    }
}
